src/Execute/ExecutionEnv.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {platform} from 'os';
+import {Command} from '../Backend/Command';
+import {Executor} from '../Backend/Executor';
+import {gToolchainEnvMap} from '../Toolchain/ToolchainEnv';
+
+// Specify Device with 'name' on `getEnableEnvList()` and `ExecutionEnv`.
+interface ExecutionEnv {
+  name(): string;
+  host(): string;
+  getEnableEnvList(): string[];
+  isAvailable(name: string): boolean;
+  getListExecutableExt(): string[];
+  getInferenceCmd(model_path: string, _options?: string[]): Command;
+}
+
+class ToolchainExecutorEnv implements ExecutionEnv {
+  envName: string;  // Specified name for Each Env
+  simulator: Executor;
+  constructor(backendName: string, sim: Executor) {
+    this.envName = backendName;
+    this.simulator = sim;
+  }
+  name(): string {
+    return this.envName;
+  }
+  host(): string {
+    return platform();
```

Comment:
I think this will be used on next step, described in https://github.com/Samsung/ONE-vscode/issues/585#issuecomment-1132476919

> Example) a toolchain is installed on local pc and it has a simulator on pc and npu runner for tv.
> * On `Target Devices`, there could be two targets
> * 1. x86/triv/simulator
> * 2. arm(or tv)/triv/one-infer

Reply:
@struss you can mix arch(), platform() and release()